code/datums/ammo/bullet/sniper.dm

```diff
@@ -107,19 +107,104 @@
+
+		if(istype(aimed_projectile.firer, /mob/living/carbon/human)) // Preps the Focused Fire marker.
+			human_firer = aimed_projectile.firer
+			focused_fire_marker_temp.color = human_firer.assigned_squad?.chat_color
+
+			if(target_mob.icon_size > world.icon_size) // Centers marker on their tile.
+				focused_fire_marker_temp.pixel_x = (target_mob.icon_size / 4)
+
+		if(istype(aimed_projectile.shot_from, /obj/item/weapon/gun/rifle/sniper/XM43E1)) // Calculates the Focus Counter.
+			var/obj/item/weapon/gun/rifle/sniper/XM43E1/amr = aimed_projectile.shot_from
+
+			old_target = amr.focused_fire_target
+
+			if(target_mob == (amr.focused_fire_target?.resolve()))
+				if(amr.focused_fire_counter < 2) // Can stack up to twice.
+					amr.focused_fire_counter += 1
+				else
+					amr.focused_fire_counter = 2
+			else // If it's a new target
+				amr.focused_fire_counter = 0 // Stacks to 0
+				if(human_firer && !(target_mob.is_dead()))
+					human_firer.client?.images -= human_firer.focused_fire_marker // Remove old marker
+					qdel(human_firer.focused_fire_marker)
+					human_firer.focused_fire_marker = focused_fire_marker_temp // Store new marker ref
+					human_firer.client?.images += focused_fire_marker_temp // Add new marker
+
+			amr_counter = amr.focused_fire_counter + 1
+			amr.focused_fire_target = WEAKREF(target_mob)
+
+		var/size_damage_mod = 0.8 // 1.8x vs Non-Xenos (225)
+		var/size_current_health_damage = 0 // % Current Health calculation, only used for Xeno calculations at the moment.
+		var/focused_fire_active = 0 // Whether to try and use focused fire calculations or not, for that kind of target.
+
+		if(isxeno(target_mob))
+			var/mob/living/carbon/xenomorph/target = target_mob
+			size_damage_mod -= 0.2 // Down to 1.6x damage, 200.
+			size_current_health_damage = 0.1 // 1.6x Damage + 10% current health (200 + 10%, 223 vs Runners)
+
+			if(target.mob_size >= MOB_SIZE_XENO && (target.caste_type != XENO_CASTE_DEFENDER))
+				size_current_health_damage += 0.1 // 1.6x Damage + 20% current health
+				focused_fire_active = 1 // Focus Fire Required. Only deals 50% bonus damage on a first Aimed Shot, then 75%, then 100%. Resets with a successful aimed shot on another target.
+
+			if(target.mob_size >= MOB_SIZE_BIG && (target.caste_type != XENO_CASTE_DEFENDER))
+				size_damage_mod -= 0.6 // Down to 1x Damage.
+				size_current_health_damage += 0.1 // 1x Damage + 30% current health.
+				focused_fire_active = 1
+				// Most T3s have around 650 to 700 HP, meaning the health modifier grants a MAXIMUM of around 195-210 damage for a total max of 320-335. This is fully focused (3 shots) and at max HP.
+				// Queen takes around 275 at max health and unfocused, 425 fully focused. Defender only takes 200 damage, even while fortified, but still causes falloff like a Big Xeno.
+				// At low health, does little more than a normal shot. Does WORSE than a normal shot if unfocused and hitting through blockers, all of which stack to reduce it.
```

Comment:
Maybe defenders should have a different message that you can't get a read on them? Also you may want to consider if you do want the defender to get the eye marker if they aren't actually getting a stack on a defender.